### PR TITLE
Add toggle component and tests

### PR DESCRIPTION
**Pull request by wasnotrice**
## Description

Adds the toggle component from [scenic_toggle](wasnotrice/scenic_toggle) as a core `Scenic.Component.Input` component.

## Motivation and Context

It's a nice component to have, and it was already on @boydm's roadmap.

## Issues

There are a few choices I made that seemed fine for a standalone component, but should be re-thunk-about in the context of Scenic core, for consistency's sake.

### Theme

I'm not totally satisfied with the handling of themes. To look "right" with the `dark` theme, the track's  "on" color has to be the theme's `thumb` color. Meanwhile, the thumb's color is the theme's `text` color. I wasn't sure about the best way to handle that, but I think it will be confusing, especially for a core component.

If you pass one of the "bootstrap" themes, then it's even weirder, because the track's "on" color should be the theme's `background` color.

### Boolean ?s

I had added `?` to the end of the boolean fields in the component state. It's idiomatic Elixir, but not idiomatic for Scenic components. Probably should match the rest of the components (i.e. should be removed from `Toggle`)

### State struct

I also made the Toggle's state into a struct. That's nice for testing, but inconsistent with the rest of the components. Since state is internal to components, it would be fine to either make Toggle state into a map like the other components, or to make the other component state into structs, to match Toggle. It shouldn't affect users either way.

## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [x] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
  not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any feature
  but make things better)

## Checklist

- [x] Check other PRs and make sure that the changes are not done yet.
- [x] The PR title is no longer than 64 characters.

### CODE REVIEW

**Comment by boydm** [comment]
Looking at it now.

**Comment by boydm** [comment]
This is great. Merging soon.

Thinking about your observations.

1) You are right, putting a ? at the end of the boolean fields is better. I'll update the other components, probably for 0.9

2) The theme thing was tricky. Somehow needed a standard way to specify a bucket of colors that gets passed around to all the components. A map with named colors seemed the easiest thing to specify. Totally get it the standard ones need more names. Also you can request that your own ones get added. Only the Components really use the theme map.

In other words, Themes was added fairly late in the game. If you have ideas on a better way to do it, I'm open to suggestions.

**Comment by wasnotrice** [comment]
....and done! Thanks :)

For themes, I think the best thing to do is add any other components that belong to Core, and then take a look at the map keys. Maybe the right names will be more obvious then.
